tests/test_rhiza/test_makefile_api.py

```
@@ -0,0 +1,267 @@
+    # The current definition in Makefile.rhiza is `pre-sync: ; @echo ...` or similar.
+    # Wait, I defined it as `pre-sync: ; @:` (single colon).
+    # So redefining it in local.mk (which is included AFTER) might trigger a warning but should work.
+
+    result = run_make(["sync"], dry_run=False)
+    # We might expect a warning about overriding commands for target `pre-sync`
+    # checking stdout/stderr for the marker
+
+    assert "[[LOCAL_PRE_SYNC]]" in result.stdout
+
+
+def test_hooks_flow(setup_api_env):
```

Ruff (0.14.13)

Unused function argument: `setup_api_env`

(ARG001)